lib/csv_helper/loader.rb

```diff
@@ -19,9 +19,22 @@ def load(file, options = {})
     def load_records(file, options)
       records = []
 
-      records = klass == Institution ? load_csv_with_version(file, records, options) : load_csv(file, records, options)
+      records = case name
+                when Program.name
+                  load_csv_with_row(file, records, options)
+                when Institution.name
+                  load_csv_with_version(file, records, options)
+                else
+                  load_csv(file, records, options)
+                end
+
       results = klass.import records, ignore: true, batch_size: Settings.active_record.batch_size.import
+
+      # using index of -1 since these rows failed during save to the table and not during after_import_validations
+      results.failed_instances.each { |result| result.errors.add(:row, -1) }
```

Comment:
This is not necessary.  All the `failed_instances` are assigned a row on line 96 of this file.
The only time that won't happen is for other `Program` validation failures because you return early from the method after running  `validator_klass.after_import_batch_validations`.
Remove this early return and those will be assigned row numbers also.  See [my PR](https://github.com/department-of-veterans-affairs/gibct-data-service/pull/513/files)'s renamed `after_import_validations_and_row_assignment` method.
If you _do_ return early, the missing row number will be [treated as 'N/A'](https://github.com/department-of-veterans-affairs/gibct-data-service/blob/edcdd3ccdfb5ee802810010d9018c8061099a72e/lib/csv_helper/shared.rb#L16), and when you sort in the controller will still appear first.  Thus the row of `-1` is still unnecessary.

Reply:
The early return in `after_import_validations` is because the goal is to either do batch validations or use Active Record Validations with a context of `after_import`

Reply:
removed line and added `.to_i` to https://github.com/department-of-veterans-affairs/gibct-data-service/blob/a20c6de4e5fa5650f46bb56f0ac99bd2dd9cd128/app/controllers/uploads_controller.rb#L23

Reply:
> The early return in `after_import_validations` is because the goal is to either do batch validations or use Active Record Validations with a context of `after_import`

I just played around with the 400K row file and observed on my machine that it did add ~10 seconds to the request if you let it continue past the current `return`.  So I agree it's good to try to reduce/eliminate that.
The reason I was advocating for it is that if a `Program` fails an import validation (e.g. `validates :facility_code, presence: true`), then when the view loads the user will see 
> Row N/A : facility_code : can't be blank

Letting the `records.each_with_index` block run would populate the row for those objects, but it's not worth the time cost.  There is another solution that I will propose in another code comment momentarily.